### PR TITLE
destination-s3: https url only in Dockerfile

### PR DESCRIPTION
**Pull request by alafanechere**
## What
https://github.com/airbytehq/airbyte/issues/22878
This connector's Dockerfile called an HTTP URL. We want to encourage the use of HTTPS only URLs in connector.
I made the change to make this QA check pass: https://github.com/airbytehq/airbyte/pull/22854

### CODE REVIEW

File: docs/integrations/destinations/s3.md [hunk 1]
```diff
@@ -166,12 +166,12 @@ A data sync may create multiple files as the output files can be partitioned by
 
 ## Supported sync modes
 
-| Feature | Support | Notes |
-| :--- | :---: | :--- |
-| Full Refresh Sync | ✅ | Warning: this mode deletes all previously synced data in the configured bucket path. |
-| Incremental - Append Sync | ✅ |  |
-| Incremental - Deduped History | ❌ | As this connector does not support dbt, we don't support this sync mode on this destination. |
-| Namespaces | ❌ | Setting a specific bucket path is equivalent to having separate namespaces. |
+| Feature                       | Support | Notes                                                                                        |
```
Comment (alafanechere) [inline docs/integrations/destinations/s3.md:169]: Markdown table formatting changes.

File: docs/integrations/destinations/s3.md [hunk 2]
```diff
@@ -128,13 +128,13 @@ For example:
...
 ↑              ↑                ↑      ↑     ↑          ↑          ↑ ↑
-|              |                |      |     |          |          | format extension
-|              |                |      |     |          |          unique incremental part id
-|              |                |      |     |          milliseconds since epoch
-|              |                |      |     upload date in YYYY_MM_DD
-|              |                |      stream name
-|              |                source namespace (if it exists)
-|              bucket path
+|             |                                 |             |                           |                          |     | format extension |
+| undefined--------- | undefined----------------------------- | undefined--------- | undefined----------------------- | undefined---------------------- | u-d | ---------------- |
```
Comment (ryankfu) [inline docs/integrations/destinations/s3.md:132]: This appears to be something the user should not be expecting, right? It seems like there's an autoformatter that believes these should be a table markdown but this is actually a rough around the edges path explanation diagram

**Review by ryankfu — APPROVED** [review]
LGTM pending path "diagram" fix

**Comment by alafanechere** [comment]
@ryankfu am I missing something in the publish flow? It fails but only publishes  [`airbyte/destination-s3/linux-amd64-0.3.20`](https://hub.docker.com/layers/airbyte/destination-s3/linux-amd64-0.3.20/images/sha256-c55636ffe8953c07743f35c83d684c4a589a88a9f53e3c50245d485e070e2451?context=explore)

**Pull request closed without merging**